packages/mdc-top-app-bar/short/foundation.ts

```diff
@@ -45,33 +47,61 @@ export class MDCShortTopAppBarFoundation extends MDCTopAppBarBaseFoundation {
       this.adapter_.addClass(cssClasses.SHORT_HAS_ACTION_ITEM_CLASS);
     }
 
-    // this is intended as the short variant must calculate if the
-    // page starts off from the top of the page.
-    this.handleTargetScroll();
+    // If initialized with SHORT_COLLAPSED_CLASS, the bar should always be collapsed
+    this.setAlwaysCollapsed(
+      this.adapter_.hasClass(cssClasses.SHORT_COLLAPSED_CLASS));
+  }
+
+  /**
+   * Set if the short top app bar should always be collapsed.
+   *
+   * When set to `true`, the bar collapses.
+   * When set to `false`, the bar will determine if it should collapse based on scroll position.
+   */
+  setAlwaysCollapsed(value: boolean) {
+    this.isAlwaysCollapsed_ = !!value;
+    if (this.isAlwaysCollapsed_) {
+      this.collapse_();
+    } else {
+      // let handleTargetScroll determine if the bar should be collapsed
+      this.handleTargetScroll();
```

Comment:
Ideally handle* methods should be called by component. This might short circuit in future if we introduce any additional logic to handleTargetScroll.

Can we abstract out the required logic from `handleTargetScroll` which can be called here?

Reply:
Done. Moved internals of `handleTargetScroll` into a private function `maybeCollapseBar_`